src/sage/topology/simplicial_complex.py

```diff
@@ -1248,13 +1351,23 @@ def _an_element_(self):
 
             sage: SimplicialComplex()._an_element_()
             ()
-            sage: simplicial_complexes.Sphere(3)._an_element_()
-            (0, 1, 2, 3)
+            sage: len(simplicial_complexes.Sphere(3)._an_element_())
+            4
         """
-        try:
-            return sorted(self.facets())[0]
-        except TypeError:
-            return self.facets()[0]
+        return self._facets[0]
+
+    def dimension(self):
+        """
+        The dimension of this simplicial complex.
+
+        EXAMPLES::
+
+            sage: simplicial_complexes.SurfaceOfGenus(20).dimension()
+            2
+            sage: simplicial_complexes.RudinBall().dimension()
+            3
+        """
+        return self._an_element_().dimension()
```

Comment:
```suggestion
        return max(facet.dimension() for facet in self._facets, default=0)
```
The test here was wrong (at best, it only works for pure simplicial complexes).